tests/foreman/api/test_activationkey.py

```diff
@@ -224,37 +177,17 @@ def test_negative_update_1(self, max_content_hosts):
         @Feature: ActivationKey
 
         """
-        try:
-            attrs = entities.ActivationKey().create_json()
-        except HTTPError as err:
-            self.fail(err)
-        activationkey = entities.ActivationKey(id=attrs['id'])
-
-        # Update the activation key with semantically incorrect values.
-        response = client.put(
-            activationkey.path(),
-            {
-                u'unlimited_content_hosts': False,
-                u'max_content_hosts': max_content_hosts
-            },
-            auth=get_server_credentials(),
-            verify=False,
-        )
-        self.assertEqual(
-            response.status_code,
-            httplib.UNPROCESSABLE_ENTITY,
-            status_code_error(
-                activationkey.path(),
-                httplib.UNPROCESSABLE_ENTITY,
-                response
-            ),
-        )
-
-        # Make sure no attributes have changed.
-        new_attrs = activationkey.read_json()
-        for attr in ('unlimited_content_hosts', 'max_content_hosts'):
-            self.assertEqual(attrs[attr], new_attrs[attr])
-        self.assertTrue(new_attrs['unlimited_content_hosts'])
+        act_key = entities.ActivationKey().create()
+        self.assertTrue(act_key.unlimited_content_hosts)
+        with self.assertRaises(HTTPError):
+            entities.ActivationKey(
+                id=act_key.id,
+                max_content_hosts=max_content_hosts,
```

Comment:
NailGun will generally send anything to the server that can be encoded as JSON. For example, you could assign any of the following values to the `max_content_hosts` field:

* `'15'`
* `'foo'`
* `15`
* `{}`
* `[]`
* `True`
* `False`
* `None`

Of course, only one of those is *semantically* valid: `15`.

An `HTTPError` is raised because the server's response contains an HTTP 4XX or 5XX status code. In other words, an error is raised not when a request is sent, but when a response is received. This approach is taken for three reasons:

* Being lax about data types lets us test the server more easily and agressively.
* It's NailGun's job to make the API easy to work with, and it's the server's job to validate data. The single responsibility principle applies to more than just object-oriented systems.
* It has already taken an enormous amount of work to make NailGun what it is, and I'm wary about accepting that bit of work too.